src/main/java/io/nats/client/impl/NatsJetStream.java

```diff
@@ -291,48 +291,47 @@ public PublishAck publish(Message message, PublishOptions options) throws IOExce
     @Override
     public CompletableFuture<PublishAck> publishAsync(Message message, PublishOptions options) {
         validateNotNull(message, "Message");
-        return publishAsync(message.getSubject(), message.getHeaders(), message.getData(), message.isUtf8mode(), options);
+        return publishAsyncInternal(message.getSubject(), message.getHeaders(), message.getData(), message.isUtf8mode(), options, null);
     }
 
-    private CompletableFuture<PublishAck> publishAsync(String subject, Headers headers, byte[] data, boolean utf8mode, PublishOptions options) {
+    private PublishAck publishSyncInternal(String subject, Headers headers, byte[] data, boolean utf8mode, PublishOptions options) throws IOException, JetStreamApiException {
+        Duration timeout = options == null ? requestTimeout : options.getStreamTimeout();
         Headers merged = mergePublishOptions(headers, options);
-        CompletableFuture<Message> future = conn.request(subject, merged, data, utf8mode, false);
+
+        Message resp = makeRequestResponseRequired(subject, merged, data, utf8mode, timeout, false);
+        return processPublishResponse(resp, options);
```

Comment:
I was handling sync messages different than async messages, when they should have been treated the same as far as 503 handling. See processPublishResponse